internal/toolexec/aspect/linkdeps/linkdeps.go

```diff
@@ -26,6 +30,22 @@ type LinkDeps struct {
 	deps map[string]struct{}
 }
 
+// FromArchive reads a link.deps file from the provided Go archive file. Returns
+// an empty LinkDeps if the archive does not contain a link.deps file.
+func FromArchive(importPath, archive string) (res LinkDeps, err error) {
+	var data io.Reader
+	data, err = readArchiveData(archive, LinkDepsFilename)
+	if err != nil {
+		return res, fmt.Errorf("reading %s from %s[%s]: %w", LinkDepsFilename, importPath, archive, err)
+	}
+	if data == nil {
+		return
+	}
+
+	log.Tracef("Found %s file in %s[%s]\n", LinkDepsFilename, importPath, archive)
+	return Read(data)
+}
```

Comment:
So `importPath` is only used to print stuff? Can we remove this from the parameters?

Reply:
The logs are much more difficult to process without it...

Reply:
`errors.Join` is go1.20 right?

Reply:
The diff tells me that this was in a fmt.Errorf higher in the stack before, let's put it where it came from and reduce the context to take to read this function